src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1143,7 +1141,14 @@ private int decodeTransfer(
                 if (!sender.fragmented)
                 {
                     assert deliveryId != NO_DELIVERY_ID; // TODO: error
-                    assert deliveryId == session.remoteDeliveryId + 1; // TODO: error
+                    final long previousDeliveryId = (session.remoteDeliveryId + 1) & 0xFFFF_FFFFL;
+                    if (deliveryId != previousDeliveryId)
+                    {
+                        server.onDecodeError(traceId, authorization, INVALID_FIELD, null);
+                        progress = limit;
+                        server.decoder = decodePlainFrame;
+                        break decode;
+                    }
```

Comment:
Alternatively, advance session remote delivery id proactively, then check for correctness and close with error if it doesn't line up.
```
session.remoteDeliveryId = (session.remoteDeliveryId + 1) & 0xFFFF_FFFFL;
if (deliveryId != session.remoteDeliveryId)
{
   ...error...
}
```